From 189972cb32b19321a84dc52e9c4fd91a8d5c9b81 Mon Sep 17 00:00:00 2001
From: Zaikuo Wang <zaikuo.wang@intel.com>
Date: Wed, 1 Apr 2015 18:18:25 +0800
Subject: [PATCH 344/347] atomisp2: correct atomisp_css_recover() flow

after isp timeout, we would call atomisp_css_recover() to recover
CSS; originally, atomisp_css_stop() be called before sensor stream off
however, the correct flow should be: sensor stream off -> css_stop -->
css_reset

Change-Id: If4f468a831147ea34f145e801641a52373baa7ba
Tracked-On: #JIMINAN-25665
Signed-off-by: Zaikuo Wang <zaikuo.wang@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/49272
Build: IMIN-legacy (SC)
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c   | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 2b90da1..1ab1fca 100755
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -1316,11 +1316,6 @@ static void __atomisp_css_recover(struct atomisp_device *isp, bool isp_timeout)
 
 		asd->streaming = ATOMISP_DEVICE_STREAMING_STOPPING;
 
-		css_pipe_id = atomisp_get_css_pipe_id(asd);
-		atomisp_css_stop(asd, css_pipe_id, true);
-
-		atomisp_acc_unload_extensions(asd);
-
 		/* stream off sensor */
 		ret = v4l2_subdev_call(
 				isp->inputs[asd->input_curr].
@@ -1329,7 +1324,13 @@ static void __atomisp_css_recover(struct atomisp_device *isp, bool isp_timeout)
 			dev_warn(isp->dev,
 					"can't stop streaming on sensor!\n");
 
+		atomisp_acc_unload_extensions(asd);
+
 		atomisp_clear_css_buffer_counters(asd);
+
+		css_pipe_id = atomisp_get_css_pipe_id(asd);
+		atomisp_css_stop(asd, css_pipe_id, true);
+
 		asd->streaming = ATOMISP_DEVICE_STREAMING_DISABLED;
 
 		asd->preview_exp_id = 1;
-- 
1.9.1

